File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/scenario/simplyteampvp/ExampleScenario.java
package com.gmail.borlandlp.minigamesdtools.arena.scenario.simplyteampvp;

import com.gmail.borlandlp.minigamesdtools.MinigamesDTools;
import com.gmail.borlandlp.minigamesdtools.arena.ArenaEventListener;
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaEventHandler;
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerJoinLocalEvent;
import com.gmail.borlandlp.minigamesdtools.arena.scenario.ScenarioAbstract;

public class ExampleScenario extends ScenarioAbstract {
    private boolean done;
    private ArenaEventListener listener;

    @Override
    public boolean isDone() {
        return this.done;
    }

    @Override
    public void update() {

    }

    @Override
    public void onInit() {

    }

    @Override
    public void beforeGameStarting() {
        this.listener = new ExampleScenarioListener(this);
        try {
            this.getArena().getEventAnnouncer().register(this.listener);
        } catch (Exception e) {
            e.printStackTrace();
        }

        try {
            MinigamesDTools.getInstance().getActivePointsAPI().activatePoint(MinigamesDTools.getInstance().getActivePointsAPI().searchPointByID("example_sphereactivepoint"));
            MinigamesDTools.getInstance().getActivePointsAPI().activatePoint(MinigamesDTools.getInstance().getActivePointsAPI().searchPointByID("example_squareactivepoint"));
            MinigamesDTools.getInstance().getActivePointsAPI().activatePoint(MinigamesDTools.getInstance().getActivePointsAPI().searchPointByID("example_livingentity"));
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public void gameEnded() {
        try {
            MinigamesDTools.getInstance().getActivePointsAPI().deactivatePoint(MinigamesDTools.getInstance().getActivePointsAPI().searchPointByID("example_sphereactivepoint"));
            MinigamesDTools.getInstance().getActivePointsAPI().deactivatePoint(MinigamesDTools.getInstance().getActivePointsAPI().searchPointByID("example_livingentity"));
            MinigamesDTools.getInstance().getActivePointsAPI().deactivatePoint(MinigamesDTools.getInstance().getActivePointsAPI().searchPointByID("example_squareactivepoint"));
        } catch (Exception e) {
            e.printStackTrace();
        }

        this.getArena().getEventAnnouncer().unregister(this.listener);
    }

    @Override
    public void beforeRoundStarting() {

    }

    @Override
    public void onRoundEnd() {

    }

    @Override
    public boolean roundMustBeEnded() {
        return false;
    }

    @Override
    public boolean gameMustBeEnded() {
        return false;
    }

    @ArenaEventHandler
    public void onEvent(ArenaPlayerJoinLocalEvent event) {

    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/scenario/simplyteampvp/ExampleScenario.kt
package com.gmail.borlandlp.minigamesdtools.arena.scenario.simplyteampvp

import com.gmail.borlandlp.minigamesdtools.MinigamesDTools.Companion.instance
import com.gmail.borlandlp.minigamesdtools.arena.ArenaEventListener
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaEventHandler
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerJoinLocalEvent
import com.gmail.borlandlp.minigamesdtools.arena.scenario.ScenarioAbstract

class ExampleScenario : ScenarioAbstract() {
    override val isDone = false
    private var listener: ArenaEventListener? = null

    override fun update() {}
    override fun onInit() {}
    override fun beforeGameStarting() {
        listener = ExampleScenarioListener(this)
        try {
            arena!!.eventAnnouncer.register(listener as ExampleScenarioListener)
        } catch (e: Exception) {
            e.printStackTrace()
        }
        try {
            instance!!.activePointsAPI!!.activatePoint(
                instance!!.activePointsAPI!!.searchPointByID(
                    "example_sphereactivepoint"
                )!!
            )
            instance!!.activePointsAPI!!.activatePoint(
                instance!!.activePointsAPI!!.searchPointByID(
                    "example_squareactivepoint"
                )!!
            )
            instance!!.activePointsAPI!!.activatePoint(
                instance!!.activePointsAPI!!.searchPointByID(
                    "example_livingentity"
                )!!
            )
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun gameEnded() {
        try {
            instance!!.activePointsAPI!!.deactivatePoint(
                instance!!.activePointsAPI!!.searchPointByID(
                    "example_sphereactivepoint"
                )!!
            )
            instance!!.activePointsAPI!!.deactivatePoint(
                instance!!.activePointsAPI!!.searchPointByID(
                    "example_livingentity"
                )!!
            )
            instance!!.activePointsAPI!!.deactivatePoint(
                instance!!.activePointsAPI!!.searchPointByID(
                    "example_squareactivepoint"
                )!!
            )
        } catch (e: Exception) {
            e.printStackTrace()
        }
        arena!!.eventAnnouncer.unregister(listener!!)
    }

    override fun beforeRoundStarting() {}
    override fun onRoundEnd() {}
    override fun roundMustBeEnded(): Boolean {
        return false
    }

    override fun gameMustBeEnded(): Boolean {
        return false
    }

    @ArenaEventHandler
    fun onEvent(event: ArenaPlayerJoinLocalEvent) {
    }
}